rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -240,6 +246,17 @@ where
     T: Parse + Serialize,
 {
     let payload_bytes = payload.payload().context("Expected payload")?;
+    let use_case: MessageUseCase = serde_json::from_slice(payload_bytes)?;
+    let killswitch_config = get_str_config("key");
+
+    if killswitch_config.is_ok() {
+        if let Some(str) = killswitch_config.unwrap() {
```

Comment:
I think it will be more readable if we moved all these lines into from_str() kind of function in the new structure that's added